### PR TITLE
generates samples csv file in background job [fixes SCI-1278]

### PR DESCRIPTION
**Pull request by ZmagoD**
https://biosistemika.atlassian.net/browse/SCI-1278

### CODE REVIEW

File: app/assets/javascripts/samples/sample_datatable.js.erb [hunk 1]
```diff
@@ -687,7 +694,14 @@ function updateButtons() {
             $("#deleteSamplesButton").removeClass("disabled");
             $("#exportSamplesButton").removeClass("disabled");
             $("#exportSamplesButton").prop("disabled",false);
-            $("#exportSamplesButton").on("click", function() { $('#form-export').submit(); });
+            $("#exportSamplesButton").on("click", function() {
```
Comment (mz3944) [inline app/assets/javascripts/samples/sample_datatable.js.erb]: We could eliminate this else statement altogether, it's so messy...

Comment (ZmagoD) [inline app/assets/javascripts/samples/sample_datatable.js.erb]: we will do that eventually when we will rewrite the front end code

Comment (mz3944) [inline app/assets/javascripts/samples/sample_datatable.js.erb]: Oh you sneaky bastard...

**Comment by mz3944** [comment]
Merge away!

**Pull request merged**